contracts/NestedFactory.sol

```diff
@@ -496,14 +480,39 @@ contract NestedFactory is INestedFactory, ReentrancyGuard, OwnableProxyDelegatio
             unchecked {
                 uint256 underSpentAmount = _amountToSpend - amounts[1];
                 if (underSpentAmount != 0) {
-                    _safeTransferWithFees(IERC20(_inputToken), underSpentAmount, _msgSender(), _nftId);
+                    SafeERC20.safeTransfer(IERC20(_inputToken), _msgSender(), underSpentAmount);
                 }
             }
         } else {
-            _safeTransferWithFees(IERC20(_inputToken), _amountToSpend, _msgSender(), _nftId);
+            SafeERC20.safeTransfer(IERC20(_inputToken), _msgSender(), _amountToSpend);
         }
     }
 
+    /// @notice Transfer fees to the FeeSplitter
+    /// @param _token The token to pay fees
+    /// @param _amount The amount transfered
+    /// @param _nftId The parent nft id (for royalties)
+    function _transferFees(
+        address _token,
+        uint256 _amount,
+        uint256 _nftId
+    ) private {
+        require(_amount != 0, "NF: ZERO_FEES");
+        // we always deposit ETH to manipulate WETH
+        if (_token == ETH) {
```

Comment:
Can't be possible since `_checkMsgValue` need the exact `msg.value` amount based on the order (extra ethers for the fees are not allowed). We must update `_checkMsgValue` to take the fees into account.

Reply:
I missed the change... Sorry!